src/components/calcite-picker/calcite-picker.tsx

```diff
@@ -0,0 +1,219 @@
+
+  /**
+   * When true, the items will be sortable via drag and drop.
+   * Only applies when mode is configuration
+   */
+  @Prop({ reflect: true }) dragEnabled = false;
+
+  @Prop({ reflect: true }) editEnabled = false; /* ignored unless mode is configuration */
+
+  /**
+   * Mode controls the presentation of the items in their selected and deselected states.
+   * Selection mode shows either radio buttons or checkboxes depending on the value of multiple
+   * Configuration mode relies on a color highlight on the edge of the item for selected
+   * Mode must be set to configuration for drag and drop behavior to work.
+   */
+  @Prop({ reflect: true }) mode: "selection" | "configuration" = "selection";
+
+  /**
+   * Multpile Works similar to standard radio buttons and checkboxes.
+   * It also affects the presented icon when in Selection mode.
+   * When true, a user can select multiple items at a time.
+   * When false, only a single item can be selected at a time,
+   * When false, selecting a new item will deselect any other selected items.
+   */
+  @Prop({ reflect: true }) multiple = false;
+
+  /**
+   * The heading label for the entire Picker.
+   * Not to be confused with the heading for an individual item or for a sub-group of items.
+   */
+  @Prop({ reflect: true }) textHeading: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @State() selectedValues = {};
+
+  items: any;
+
+  lastSelectedItem = null;
+
+  guid = `calcite-picker-${guid()}`;
+
+  observer = new MutationObserver(() => this.setupItems());
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.setupItems();
+  }
+
+  componentDidLoad() {
+    this.observer.observe(this.el, { childList: true, subtree: true });
+  }
+
+  componentDidUnload() {
+    this.observer.disconnect();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerSelectionChange: EventEmitter;
+
+  @Listen("calcitePickerItemToggled") calcitePickerItemToggledHandler(event) {
+    event.stopPropagation(); // private event
+    const { item, value, selected, shiftPressed } = event.detail;
+    if (selected) {
+      if (this.multiple && shiftPressed && this.lastSelectedItem) {
+        const start = this.items.indexOf(this.lastSelectedItem);
```

Comment:
Since `this.items` is used a few times, could you use a local variable for readability? May also apply to other properties.